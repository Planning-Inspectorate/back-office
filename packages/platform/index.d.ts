import './config/dotenv';
import './http/ttl';
import './util/CryptoUtils';
import './util/date';
import './util/filter';
import './util/url';
import './validators/date';
import './validators/postcode';

declare module '@pins/platform' {
	export * from './config/dotenv';
	export * from './http/ttl';
	export * from './util/CryptoUtils';
	export * from './util/date';
	export * from './util/filter';
	export * from './util/url';
	export * from './validators/date';
	export * from './validators/postcode';
}
