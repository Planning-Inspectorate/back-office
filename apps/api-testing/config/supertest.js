// @ts-nocheck
import supertest from 'supertest';
import config from './config.cjs';
import { Logger } from '../utils/logger.js';

const logger = new Logger();
logger.log('green', '************* Tests starting ****************');
logger.log('cyan', `Running tests against: ${config.baseUrl}`);
logger.log('green', '*********************************************');

const request = supertest(config.baseUrl);

export default request;
