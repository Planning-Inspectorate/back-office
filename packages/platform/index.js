export * from './http/hooks.js';
export * from './testing/assets/assets.js';
export * from './testing/fake.js';
export * from './testing/html-parser.js';
export * from './utils/date.js';
export * from './utils/dotenv.js';
export * from './utils/filter.js';
export * from './validators/date.js';
export * from './validators/postcode.js';
