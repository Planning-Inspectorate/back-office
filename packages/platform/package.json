{
  "name": "@pins/platform",
  "version": "0.0.0",
  "type": "module",
  "main": "index.js",
  "private": true,
  "engines": {
    "node": ">=16.14.0"
  },
  "scripts": {
    "clean": "rimraf .turbo node_modules dist",
    "lint:js": "npx eslint .",
    "lint:js:fix": "npx eslint . --fix",
    "tscheck": "npx tsc -p jsconfig.json --maxNodeModuleJsDepth 0"
  },
  "dependencies": {
    "date-fns": "2.28.0",
    "dotenv": "16.0.0",
    "got": "^12.5.2",
    "html": "^1.0.0",
    "lodash-es": "^4.17.21",
    "node-html-parser": "^5.3.3",
    "randexp": "^0.5.3"
  },
  "devDependencies": {
    "@types/html": "^1.0.1",
    "@types/lodash-es": "^4.17.6",
    "type-fest": "^2.12.2"
  }
}
