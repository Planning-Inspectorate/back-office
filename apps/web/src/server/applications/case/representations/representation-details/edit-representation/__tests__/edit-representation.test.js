import { parseHtml } from '@pins/platform';
import nock from 'nock';
import supertest from 'supertest';
import { createTestEnvironment } from '../../../../../../../../testing/index.js';
import { representationDetailsFixture } from '../../__fixtures__/representation-details.fixture.js';

const { app, installMockApi, teardown } = createTestEnvironment();
const request = supertest(app);

const mockCaseReference = { title: 'mock title', status: 'in test', reference: 'mock reference' };

const setupNocks = () => {
	nock('http://test/').get('/applications/1').reply(200, mockCaseReference);

	nock('http://test/')
		.get('/applications/1/representations/1')
		.reply(200, representationDetailsFixture)
		.persist();

	nock('http://test/')
		.patch('/applications/1/representations/1/edit')
		.reply(200, { message: 'patched' })
		.persist();
};

describe('Edit Representation Page', () => {
	beforeEach(installMockApi);
	afterEach(teardown);
	afterAll(() => nock.cleanAll());

	const baseUrl =
		'/applications-service/case/1/relevant-representations/1/representation-details/edit-representation';

	describe('GET', () => {
		beforeEach(async () => {
			setupNocks();
			await request.get('/applications-service/');
		});

		it('renders the edit representation page', async () => {
			const response = await request.get(baseUrl);
			const element = parseHtml(response.text);
			expect(element.innerHTML).toMatchSnapshot();
		});
	});

	describe('POST', () => {
		beforeEach(async () => {
			setupNocks();
			await request.get('/applications-service/');
		});

		it('shows error when editedRepresentation is missing', async () => {
			const response = await request.post(baseUrl).send({});
			const element = parseHtml(response.text);
			expect(element.innerHTML).toMatchSnapshot();
		});

		it('patches and redirects on valid submission', async () => {
			const response = await request.post(baseUrl).send({
				editedRepresentation: 'Test representation',
				editNotes: 'Some notes'
			});
			expect(response.headers.location).toBe(
				'/applications-service/case/1/relevant-representations/1/representation-details'
			);
		});
	});
});
