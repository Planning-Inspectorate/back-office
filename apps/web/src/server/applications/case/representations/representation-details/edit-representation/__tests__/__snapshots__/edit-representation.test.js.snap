// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit Representation Page GET renders the edit representation page 1`] = `
"<main class="govuk-main-wrapper" id="main-content">
    <div class="govuk-grid-row">
        <div class="govuk-grid-column-two-thirds">
            <form method="post" novalidate>
                <h1 class="govuk-heading-l govuk-!-margin-bottom-2">Mrs Sue</h1><strong class="govuk-heading-s"><strong class="govuk-tag govuk-tag--">Awaiting review</strong></strong>
                <p                 class="govuk-body-s"><strong>Project name:</strong> mock title</p>
                    <div class="govuk-form-group">
                        <label class="govuk-label govuk-label--s" for="edited-representation">Representation content</label>
                        <textarea class="govuk-textarea" id="edited-representation"
                        name="editedRepresentation" rows="7">Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo
                            ligula eget dolor. Aenean massa. Cum sociis natoque penatibus et magnis
                            dis parturient montes, nascetur ridiculus mus. Donec quam felis, ultricies
                            nec, pellentesque eu, pretium quis, sem. Nulla con.</textarea>
                    </div>
                    <div class="govuk-form-group">
                        <label class="govuk-label govuk-label--s" for="edit-notes">Notes</label>
                        <textarea class="govuk-textarea" id="edit-notes" name="editNotes"
                        rows="2"></textarea>
                    </div>
                    <div class="govuk-warning-text"><span class="govuk-warning-text__icon" aria-hidden="true">!</span>
                        <strong                         class="govuk-warning-text__text"><span class="govuk-visually-hidden">Warning</span> Editing a representation
                            will remove any current redactions</strong>
                    </div>
                    <button type="submit" class="govuk-button govuk-!-margin-top-3" data-module="govuk-button">Save and return</button>
            </form>
        </div>
    </div>
</main>"
`;

exports[`Edit Representation Page POST shows error when editedRepresentation is missing 1`] = `
"<main class="govuk-main-wrapper" id="main-content">
    <div class="govuk-grid-row">
        <div class="govuk-grid-column-full">
            <div class="govuk-error-summary" data-module="govuk-error-summary">
                <div role="alert">
                    <h2 class="govuk-error-summary__title"> There is a problem</h2>
                    <div class="govuk-error-summary__body">
                        <ul class="govuk-list govuk-error-summary__list">
                            <li><a href="#editedRepresentation">Enter representation content</a>
                            </li>
                        </ul>
                    </div>
                </div>
            </div>
        </div>
    </div>
    <div class="govuk-grid-row">
        <div class="govuk-grid-column-two-thirds">
            <form method="post" novalidate>
                <h1 class="govuk-heading-l govuk-!-margin-bottom-2">Mrs Sue</h1><strong class="govuk-heading-s"><strong class="govuk-tag govuk-tag--">Awaiting review</strong></strong>
                <p                 class="govuk-body-s"><strong>Project name:</strong> mock title</p>
                    <div class="govuk-form-group govuk-form-group--error">
                        <label class="govuk-label govuk-label--s" for="edited-representation">Representation content</label>
                        <p id="edited-representation-error" class="govuk-error-message"><span class="govuk-visually-hidden">Error:</span> Enter representation
                            content</p>
                        <textarea class="govuk-textarea govuk-textarea--error" id="edited-representation"
                        name="editedRepresentation" rows="7" aria-describedby="edited-representation-error"></textarea>
                    </div>
                    <div class="govuk-form-group">
                        <label class="govuk-label govuk-label--s" for="edit-notes">Notes</label>
                        <textarea class="govuk-textarea" id="edit-notes" name="editNotes"
                        rows="2"></textarea>
                    </div>
                    <div class="govuk-warning-text"><span class="govuk-warning-text__icon" aria-hidden="true">!</span>
                        <strong                         class="govuk-warning-text__text"><span class="govuk-visually-hidden">Warning</span> Editing a representation
                            will remove any current redactions</strong>
                    </div>
                    <button type="submit" class="govuk-button govuk-!-margin-top-3" data-module="govuk-button">Save and return</button>
            </form>
        </div>
    </div>
</main>"
`;
